### PR TITLE
Saryn Mooney

### PR DESCRIPTION
**Pull request by sarynm12**

### CODE REVIEW

File: lib/author.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class Author
+  attr_reader :first_name, :last_name, :books
+
+  def initialize(info)
+    @first_name = info[:first_name]
+    @last_name = info[:last_name]
```
Comment (timomitchel) [inline lib/author.rb:6]: Do we need to store these as instance variables? What if we did the string concatenation inside initialize and then had a `@name` instance variable.

File: lib/author.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    @books = []
+  end
+
+  def name
+    @first_name + " " + @last_name
+  end
+
+  def write(title, publication_date)
```
Comment (timomitchel) [inline lib/author.rb:14]: Nice job with this method.

File: lib/book.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+  def initialize(info)
+    @author_first_name = info[:author_first_name]
+    @author_last_name = info[:author_last_name]
+    @title = info[:title]
+    @publication_date = info[:publication_date]
+  end
+
+  def author
+    @author = @author_first_name + " " + @author_last_name
```
Comment (timomitchel) [inline lib/book.rb:12]: This `@author` should be removed.

File: lib/library.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
+class Library
+  attr_reader :name, :books, :authors, :checked_out_books
+
+  def initialize(name)
+    @name = name
+    @authors = []
+    @checked_out_books = []
+  end
+
+  def books
```
Comment (timomitchel) [inline lib/library.rb:10]: Great job with this method.

File: lib/library.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+      author.books
+    end
+  end
+
+  def add_author(author)
+    @authors << author
+  end
+
+  def publication_time_frame_for(author)
+    result = {:start => author.books[0].publication_year,
```
Comment (timomitchel) [inline lib/library.rb:21]: Not quite dynamic enough. We need a max and a min of each author's book publication date. There are a number of ways to approach this, but I prefer sorting the books by the publication year and then taking the first and last elements of that array.

File: lib/library.rb [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+              :end => author.books[1].publication_year}
+    # elsif books.count < 1
+    #   result = {:start => author.books[0].publication_year,
+    #           :end => author.books[0].publication_year}
+    # end
+  end
+
+  def checkout(book)
```
Comment (timomitchel) [inline lib/library.rb:29]: On the right track here, but missing some of the functionality.

File: test/author_test.rb [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    assert_equal "Jane Eyre", jane_eyre.title
+  end
+
+  def test_written_books_get_added_to_books
+    charlotte_bronte = Author.new({first_name: "Charlotte", last_name: "Bronte"})
+    jane_eyre = charlotte_bronte.write("Jane Eyre", "October 16, 1847")
+    assert_equal [jane_eyre], charlotte_bronte.books
+    villette = charlotte_bronte.write("Villette", "1853")
+    assert_equal [jane_eyre, villette], charlotte_bronte.books
+  end
```
Comment (timomitchel) [inline test/author_test.rb:36]: Nice testing throughout.

File: test/library_test.rb [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+    villette = charlotte_bronte.write("Villette", "1853")
+    harper_lee = Author.new({first_name: "Harper", last_name: "Lee"})
+    mockingbird = harper_lee.write("To Kill a Mockingbird", "July 11, 1960")
+    assert_equal false, dpl.checkout(mockingbird)
+    assert_equal false, dpl.checkout(jane_eyre)
+    dpl.add_author(charlotte_bronte)
+    dpl.add_author(harper_lee)
+    assert_equal true, dpl.checkout(jane_eyre)
+    assert_equal true, dpl.checkout(villette)
+  end
```
Comment (timomitchel) [inline test/library_test.rb:75]: Nice robust testing of this both boolean literals here.

**Comment by sarynm12** [comment]
Thanks Tim! I appreciate the feedback and encouragement. Definitely feeling better about things this go around. As always, thank you for all that you do!

**Pull request closed without merging**